### ISSUE TITLE
breeze.js doesn't seem to support saving of changes to non-scalar primitive types

### ISSUE THREAD

**Issue by maximumduncan** [Breeze/breeze.js#84]
I've noticed that there are a couple of github issues lying about relating to non-scalar complex types, but I think I've found a completely separate issue relating to mutating non-scalar *primitive* data properties (those properties which essentially map to arrays of string, int etc.).

You can mutate these arrays, and breeze successfully knows that the entity has changed but if you try and reject or accept those changes nothing happens. It seems that breeze.js is missing functionality at these points.

I've successfully managed to patch in the required behaviour which you can view using the link below, but I'd thought I'd ask whether this is a known limitation, or whether I'm missing something?

https://github.com/edigitalresearch/breeze.js/compare/master...edigitalresearch:support-saving-nonscalar-primitives?expand=1

**Comment by wardbell** [comment]
You are correct that we've been slow to get to the complex type and array issues. We've prioritized other tasks. I have some confidence that well investigate these soon.

Thanks for your efforts on this one and thanks for your patience.
